README.md

```diff
@@ -1,2 +1,152 @@
+    * Documentation: https://cython.readthedocs.io/en/latest/
+* Google `protobuf` and `protobuf-c`: Google `protobuf` is the library that we use to serialize our messages between Runtime and Shepherd, and Runtime and Dawn. The brief explanation of how it works is this: the user defines the structure of a message in "protobuf language", and saves it as a `.proto` file. Google's protobuf compiler will then take that `.proto` file and generate code that can be used in a desired target language to serialize and deserialize ("pack" and "unpack" in the language of protobufs) messages of that type. Since Google's protobuf compiler does not have native support for C, we need to use the third party library `protobuf-c` to generate C code. (But `protobuf-c` makes use of Google's library, so we still need it).
+    * `protobuf` documentation: https://developers.google.com/protocol-buffers/docs/proto3
+	* `protobuf` Github: https://github.com/protocolbuffers/protobuf
+	* `protobuf-c` Github: https://github.com/protobuf-c/protobuf-c
+
+Runtime uses the folllowing C and POSIX standard library headers, in rough order from most commonly used to least commonly used (these all do not require additional installation before they can be used on any POSIX-compliant systems):
+
+* `stdio.h`: for standard input/output (ex. `printf`)
+* `stdlib.h`: C standard utility functions (ex. `sleep`, `malloc`, `free`)
+* `stdint.h`: C standard integer types (ex. `uint16_t`, `uint32_t`)
+* `pthread.h`: for threading functions (ex. `pthread_create`, `pthread_mutex_lock`, `pthread_cond_signal`)
+* `unistd.h`: for miscellaneous input/output functions and functions (ex. `read`, `write`, `open`, `close`, `SEEK_END`)
+* `errno.h`: for a bunch of error constants and functions (ex. `EEXIST`, `EWOULDBLOCK`, `EINTR`)
+* `arpa/inet.h`: for socket functions (ex. `inet_addr`, `bind`, `listen`, `accept`, `socket`)
+* `netinet/in.h`: for socket address structures (ex. `struct sockaddr`, `struct sockaddr_in`)
+* `sys/types.h`: for some system-defined types for holding various objects (ex. `sem_t` for semaphores)
+* `semaphore.h`: for semaphores (ex. `sem_wait`, `sem_post`)
+* `sys/mman.h`: for memory mapping used in shared memory (ex. `mmap`)
+* `sys/time.h` and `time.h`: for measuring time (ex. `time_t`, `ctime`)
+* `Python.h`: for calling Python functions from C used in executor (ex. `Py_XDECREF`, `PyObject_CallObject`)
+* `signal.h`: for signal handlers and signal constants (ex. `signal`, `SIGINT`)
+* `string.h`: for string manipulation (ex. `strcpy`, `memset`)
+* `stdarg.h`: for variable-length arguments used in `logger` (ex. `va_args`)
+* `sys/stat.h`: for working with unconventional files (ex. `mkfifo`)
+* `sys/wait.h`: for wait functions used in executor to wait for processes (ex. `waitpid`)
+* `fcntl.h`: for constants used in opening and closing files (ex. `O_RDONLY`, `O_WRONLY`)
+
```

Comment:
I would like to add a section here, can be TODO for now, called ## Building Runtime

Reply:
👌

Reply:
I mean unless you include building as part of Running Runtime. But there needs to be instructions somewhere on how to build the project.

Reply:
This is already found in the READMEs for `dev_handler`, `net_handler`, and `executor` right? Like if you want to build the project, you install dependencies (included in the "Dependencies" section of this README), go into each of those folders and `make` each part, and then follow instructions in "Running Runtime" to run it

Reply:
Uhhh that seems like a lot of work. There should be a master Makefile in the root directory that can make everything for you with just `make`.

Reply:
Ok, good point. Will add a Building Runtime section~~